base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -120,9 +141,12 @@ where T: BlockchainBackend
                 Ok(NodeCommsResponse::TransactionOutputs(utxos))
             },
             NodeCommsRequest::FetchBlocks(block_nums) => {
+                let mut block_nums = block_nums.clone();
+                block_nums.truncate(min(block_nums.len(), self.config.max_block_response_batch_size.clone()));
```

Comment:
```suggestion
                let batch_size = max(1, self.config.max_block_response_batch_size);
                let batch_size = min(block_nums.len(), batch_size);
                block_nums.truncate(batch_size);
```
Assuming this will eventually be user-specifed, prevent a user doing weird things like setting the value to 0 or negative

Reply:
That is a nice improvement.